components/eamxx/src/diagnostics/tests/number_paths_tests.cpp

```diff
@@ -0,0 +1,243 @@
+  params.set<std::string>("Number Kind", "Foo");
+  REQUIRE_THROWS(diag_factory.create("NumberPath", comm,
+                                     params));  // Invalid 'Number Kind'
+
+  // Liquid
+  params.set<std::string>("Number Kind", "Liq");
+  auto diag_liq = diag_factory.create("NumberPath", comm, params);
+  diag_liq->set_grids(gm);
+  diags.emplace("lnp", diag_liq);
+  // Ice
+  params.set<std::string>("Number Kind", "Ice");
+  auto diag_ice = diag_factory.create("NumberPath", comm, params);
+  diag_ice->set_grids(gm);
+  diags.emplace("inp", diag_ice);
+  // Rain
+  params.set<std::string>("Number Kind", "Rain");
+  auto diag_rain = diag_factory.create("NumberPath", comm, params);
+  diag_rain->set_grids(gm);
+  diags.emplace("rnp", diag_rain);
+
+  // Set the required fields for the diagnostic.
+  std::map<std::string, Field> input_fields;
+  for(const auto &dd : diags) {
+    const auto &diag = dd.second;
+    for(const auto &req : diag->get_required_field_requests()) {
+      if(input_fields.find(req.fid.name()) == input_fields.end()) {
+        Field f(req.fid);
+        f.allocate_view();
+        f.get_header().get_tracking().update_time_stamp(t0);
+        input_fields.emplace(f.name(), f);
+      }
+      const auto &f = input_fields.at(req.fid.name());
+      diag->set_required_field(f.get_const());
+      REQUIRE_THROWS(diag->set_computed_field(f));
+    }
+    // Initialize the diagnostic
+    diag->initialize(t0, RunType::Initial);
+  }
+
+  // Run tests
+  {
+    // Construct random data to use for test
+    // Get views of input data and set to random values
+    const auto &qc_f          = input_fields.at("qc");
+    const auto &qc_v          = qc_f.get_view<Real **>();
+    const auto &nc_f          = input_fields.at("nc");
+    const auto &nc_v          = nc_f.get_view<Real **>();
+    const auto &qi_f          = input_fields.at("qi");
+    const auto &qi_v          = qi_f.get_view<Real **>();
+    const auto &ni_f          = input_fields.at("ni");
+    const auto &ni_v          = ni_f.get_view<Real **>();
+    const auto &qr_f          = input_fields.at("qr");
+    const auto &qr_v          = qr_f.get_view<Real **>();
+    const auto &nr_f          = input_fields.at("nr");
+    const auto &nr_v          = nr_f.get_view<Real **>();
+    const auto &pseudo_dens_f = input_fields.at("pseudo_density");
+    const auto &pseudo_dens_v = pseudo_dens_f.get_view<Real **>();
+    for(int icol = 0; icol < ncols; icol++) {
+      const auto &qc_sub = ekat::subview(qc_v, icol);
+      const auto &nc_sub = ekat::subview(nc_v, icol);
+      const auto &qi_sub = ekat::subview(qi_v, icol);
+      const auto &ni_sub = ekat::subview(ni_v, icol);
+      const auto &qr_sub = ekat::subview(qr_v, icol);
+      const auto &nr_sub = ekat::subview(nr_v, icol);
+      const auto &dp_sub = ekat::subview(pseudo_dens_v, icol);
+      ekat::genRandArray(pseudo_density, engine, pdf_pseudodens);
```

Comment:
Why not calling `genRandArray` directly on `xyz_sub`? Seems less lines of code, and easier to maintain/follow.